src/checkbox/tests/unit/Checkbox.ts

```diff
@@ -1,151 +1,242 @@
 import * as registerSuite from 'intern!object';
 import * as assert from 'intern/chai!assert';
-import { VNode } from '@dojo/interfaces/vdom';
-import Checkbox, { Mode } from '../../Checkbox';
+import * as Test from 'intern/lib/Test';
+
+import has from '@dojo/has/has';
+import harness, { assignChildProperties, assignProperties, Harness, replaceChild } from '@dojo/test-extras/harness';
+import { v, w } from '@dojo/widget-core/d';
+import { HNode } from '@dojo/widget-core/interfaces';
+
+import Checkbox, { CheckboxProperties, Mode } from '../../Checkbox';
+import Label, { LabelProperties } from '../../../label/Label';
 import * as css from '../../styles/checkbox.m.css';
 
+let widget: Harness<CheckboxProperties, typeof Checkbox>;
+
 registerSuite({
 	name: 'Checkbox',
 
-	construction() {
-		const checkbox = new Checkbox();
-		checkbox.setProperties({
-			checked: true
-		});
+	beforeEach() {
+		widget = harness(Checkbox);
+	},
 
-		assert.isTrue(checkbox.properties.checked);
+	afterEach() {
+		widget.destroy();
 	},
 
-	'default node attributes'() {
-		const checkbox = new Checkbox();
-		const vnode = <VNode> checkbox.__render__();
-		const inputNode = vnode.children![0].children![0];
+	simple() {
+		const children: HNode[] = [
+			v('div', {
+				classes: widget.classes(css.inputWrapper)
+			}, [
+				null,
+				null,
+				v('input', {
+					'aria-describedby': undefined,
+					'aria-invalid': null,
+					'aria-readonly': null,
+					bind: widget,
+					checked: false,
+					classes: widget.classes(css.input),
+					disabled: undefined,
+					name: undefined,
+					onblur: widget.listener,
+					onchange: widget.listener,
+					onclick: widget.listener,
+					onfocus: widget.listener,
+					onmousedown: widget.listener,
+					onmouseup: widget.listener,
+					ontouchstart: widget.listener,
+					ontouchend: widget.listener,
+					ontouchcancel: widget.listener,
+					readOnly: undefined,
+					required: undefined,
+					type: 'checkbox',
+					value: undefined
+				})
+			])
+		];
+
+		widget.setChildren(children);
 
-		assert.strictEqual(inputNode.vnodeSelector, 'input');
-		assert.strictEqual(inputNode.properties!.type, 'checkbox');
-		assert.strictEqual(inputNode.properties!.checked, false);
+		widget.expectRender(v('div', {
+			classes: widget.classes(css.root)
+		}, children));
+
+		assignProperties(children[0], {
+			classes: widget.classes(css.inputWrapper)
+		});
+		replaceChild(children[0], 0, v('div', { classes: widget.classes(css.onLabel) }, [ null ]));
+		replaceChild(children[0], 1, v('div', { classes: widget.classes(css.offLabel) }, [ null ]));
+
+		widget.setProperties({
+			invalid: false,
+			mode: Mode.toggle
+		});
+		widget.setChildren(children);
+		widget.expectRender(v('div', {
```

Comment:
It will be helpful if someone can figure out why this test fails. The one above on line 62 is fine, but after manipulating a few things and calling `expectRender` again, it fails.